source/dofs/dof_renumbering.cc

```diff
@@ -723,12 +723,26 @@ namespace DoFRenumbering
         renumbering, start, end, component_order_arg, true);
     (void)result;
 
-    Assert(result == 0 || result == dof_handler.n_dofs(level),
```

Comment:
... and fix this assert so that `DoFRenumber::component_wise()` works in parallel.